### PR TITLE
operator 3scale-community-operator (0.11.0)

### PR DESCRIPTION
**Pull request by valerymo**
Thanks for submitting your Operator. Please check the below list before you create your Pull Request.

### Updates to existing Operators

* [x] Is your new CSV pointing to the previous version with the replaces property if you chose replaces-mode via the updateGraph property in ci.yaml?
* [x] Is your new CSV referenced in the appropriate channel defined in the package.yaml or annotations.yaml ?
* [x] Have you tested an update to your Operator when deployed via OLM?
* [x] Is your submission signed?

### Your submission should not

* [x] Add more than one operator bundle per PR
* [x] Modify any operator
* [x] Rename an operator
* [x] Modify any files outside the above mentioned folders
* [x] Contain more than one commit. **Please squash your commits.**

### Operator Description must contain (in order)

1. [x] Description of the managed Application and where to find more information
2. [x] Features and capabilities of your Operator and how to use it
3. [x] Any manual steps about potential pre-requisites for using your Operator

### Operator Metadata should contain

* [x] Human readable name and 1-liner description about your Operator
* [x] Valid [category name](https://github.com/redhat-openshift-ecosystem/community-operators-prod/blob/main/docs/packaging-operator.md#categories)<sup>1</sup>
* [x] One of the pre-defined [capability levels](https://github.com/operator-framework/operator-courier/blob/4d1a25d2c8d52f7de6297ec18d8afd6521236aa2/operatorcourier/validate.py#L556)<sup>2</sup>
* [x] Links to the maintainer, source code and documentation
* [x] Example templates for all Custom Resource Definitions intended to be used
* [x] A quadratic logo

Remember that you can preview your CSV [here](https://operatorhub.io/preview).

--

<sup>1</sup> If you feel your Operator does not fit any of the pre-defined categories, file an issue against this repo and explain your need

<sup>2</sup> For more information see [here](https://sdk.operatorframework.io/docs/overview/#operator-capability-level)

### CODE REVIEW

**Review by eguzki — CHANGES_REQUESTED** [review]
wrong image references 

```
               - name: RELATED_IMAGE_APICAST
                  value: quay.io/3scale/3scale213:apicast-3scale-2.13.1-GA
                - name: RELATED_IMAGE_SYSTEM
                  value: quay.io/3scale/3scale213:porta-3scale-2.13.1-GA
                - name: RELATED_IMAGE_ZYNC
                  value: quay.io/3scale/3scale213:zync-3scale-2.13.1-GA
```

**Comment by valerymo** [comment]
> wrong image references
> 
> ```
>                - name: RELATED_IMAGE_APICAST
>                   value: quay.io/3scale/3scale213:apicast-3scale-2.13.1-GA
>                 - name: RELATED_IMAGE_SYSTEM
>                   value: quay.io/3scale/3scale213:porta-3scale-2.13.1-GA
>                 - name: RELATED_IMAGE_ZYNC
>                   value: quay.io/3scale/3scale213:zync-3scale-2.13.1-GA
> ```

Thank you! Yes, it was draft. Now - updated.

**Comment by valerymo** [comment]
> 3scale-2.13.1-GA

Updated. Thank you

File: operators/3scale-community-operator/0.11.0/bundle.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=3scale-community-operator
+LABEL operators.operatorframework.io.bundle.channels.v1=threescale-2.14
+LABEL operators.operatorframework.io.bundle.channel.default.v1=threescale-2.14
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.2.0
+LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
+LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v2
+LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
+LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
+LABEL com.redhat.openshift.versions=v4.9
```
Comment (MStokluska) [inline operators/3scale-community-operator/0.11.0/bundle.Dockerfile:14]: is this correct? We want this to be available on 4.9?

File: operators/3scale-community-operator/0.11.0/manifests/3scale-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,1042 @@
...
+    name: Eguzki Astiz Lezaun
+  - email: mstoklus@redhat.com
+    name: Michal Stokluska
+  - email: aucunnin@redhat.com
+    name: Austin Cunningham
+  maturity: stable
+  provider:
+    name: Red Hat
+  version: 0.11.0
+  replaces: 3scale-community-operator.v0.10.1
```
[the remark this answers is not in the document]
Comment (MStokluska) [reply operators/3scale-community-operator/0.11.0/manifests/3scale-operator.clusterserviceversion.yaml:1042]: Although I see this is how it was also done for 2.12 > 2.13 move..

[the remark this answers is not in the document]
Comment (valerymo) [reply operators/3scale-community-operator/0.11.0/manifests/3scale-operator.clusterserviceversion.yaml:1042]: Thanks for comment, yes, in previouse release it's similar:
  version: 0.10.1
  replaces: 3scale-community-operator.v0.9.0

File: operators/3scale-community-operator/0.11.0/bundle.Dockerfile (hunk 1 above)
Comment (valerymo) [reply operators/3scale-community-operator/0.11.0/bundle.Dockerfile]: I removed this line: LABEL com.redhat.openshift.versions=v4.9. It doesn't exist in 3scale-operator bundle.Dockerfile. 
Thank you

**Comment by valerymo** [comment]
> RELATED_IMAGE_APICAST

Done

**Comment by eguzki** [comment]
From the description, I see "New Submissions". This is not a new submission, this is a "Updates to existing Operators" kind of submit. 

The `metadata/annotations.yaml` file is missing `com.redhat.openshift.versions`

It should be like this

```yaml
annotations:
  operators.operatorframework.io.bundle.channel.default.v1: threescale-2.14
  operators.operatorframework.io.bundle.channels.v1: threescale-2.14
  operators.operatorframework.io.bundle.manifests.v1: manifests/
  operators.operatorframework.io.bundle.mediatype.v1: registry+v1
  operators.operatorframework.io.bundle.metadata.v1: metadata/
  operators.operatorframework.io.bundle.package.v1: 3scale-community-operator
  operators.operatorframework.io.metrics.builder: operator-sdk-v1.2.0
  operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
  operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v2
  operators.operatorframework.io.test.config.v1: tests/scorecard/
  operators.operatorframework.io.test.mediatype.v1: scorecard+v1
  com.redhat.openshift.versions: v4.11-v4.15
```

**Comment by valerymo** [comment]
> after applying the requested changes, make the PR ready for review

Changes done.

**Comment by valerymo** [comment]
Hi @eguzki , following test failed `evaluate-preflight-result`. Could you please help. Log fragment is below. Thank you.
```
Not all preflight tests passed.
{
  "image": "quay.io/community-operator-pipeline-prod/3scale-community-operator:0.11.0",
  "passed": false,
  "certification_hash": "d41d8cd98f00b204e9800998ecf8427e",
  "test_library": {
    "name": "github.com/redhat-openshift-ecosystem/openshift-preflight",
    "version": "1.9.2",
    "commit": "4a0c6b9b74de9a4ac8a812f5e1f4aa9144755823"
  },
  "results": {
    "passed": [
      {
        "name": "ScorecardBasicSpecCheck",
        "elapsed_time": 11636,
        "description": "Check to make sure that all CRs have a spec block."
      },
      {
        "name": "ScorecardOlmSuiteCheck",
        "elapsed_time": 2606,
        "description": "Operator-sdk scorecard OLM Test Suite Check"
      }
    ],
    "failed": [],
    "errors": [
      {
        "name": "DeployableByOLM",
        "elapsed_time": 183260,
        "description": "Checking if the operator could be deployed by OLM",
        "help": "It is required that your operator could be deployed by OLM"
      }
    ]
  }
}
failure

```

**Comment by eguzki** [comment]
content is correct. https://github.com/3scale/3scale-operator/pull/979 + patch

```diff
diff --git a/bundle/manifests/3scale-operator.clusterserviceversion.yaml b/bundle/manifests/3scale-operator.clusterserviceversion.yaml
index 0155950d..d451478d 100644
--- a/bundle/manifests/3scale-operator.clusterserviceversion.yaml
+++ b/bundle/manifests/3scale-operator.clusterserviceversion.yaml
@@ -217,7 +217,7 @@ metadata:
     operatorframework.io/arch.amd64: supported
     operatorframework.io/arch.ppc64le: supported
     operatorframework.io/arch.s390x: supported
-  name: 3scale-operator.v0.11.0
+  name: 3scale-community-operator.v0.11.0
   namespace: placeholder
 spec:
   apiservicedefinitions: {}
diff --git a/bundle/metadata/annotations.yaml b/bundle/metadata/annotations.yaml
index a4ab00c0..7ace76b5 100644
--- a/bundle/metadata/annotations.yaml
+++ b/bundle/metadata/annotations.yaml
@@ -4,9 +4,10 @@ annotations:
   operators.operatorframework.io.bundle.manifests.v1: manifests/
   operators.operatorframework.io.bundle.mediatype.v1: registry+v1
   operators.operatorframework.io.bundle.metadata.v1: metadata/
-  operators.operatorframework.io.bundle.package.v1: 3scale-operator
+  operators.operatorframework.io.bundle.package.v1: 3scale-community-operator
   operators.operatorframework.io.metrics.builder: operator-sdk-v1.2.0
   operators.operatorframework.io.metrics.mediatype.v1: metrics+v1
   operators.operatorframework.io.metrics.project_layout: go.kubebuilder.io/v2
   operators.operatorframework.io.test.config.v1: tests/scorecard/
   operators.operatorframework.io.test.mediatype.v1: scorecard+v1
+  com.redhat.openshift.versions: v4.11-v4.15
```

**Review by eguzki — APPROVED** [review]

**Comment by Allda** [comment]
It looks like the preflight test is using a package name for `CatalogSource` name when trying to install the operator on a cluster. @acornett21 would you have any suggestions to @eguzki on how to work around this issue? Can we start using a CatalogSource that doesn't use the package as a name to avoid these kinds of issues?

**Comment by acornett21** [comment]
This is a K8s validation error, I'm not sure which version of K8s this went into, but since this operator hasn't been released in over a year, it would be really hard to track down where/when this was changed. I think the operator name and package name would have to change, so probably would have to be a new operator folder in this repo and treated as net new. Something like `three-scale-operator`

**Comment by eguzki** [comment]
This is very inconvenient. But if there is no other way :shrug:

**Comment by MStokluska** [comment]
hey @acornett21 @Allda 
It seems that the issue (as mentioned above) is caused directly by metadata.name of catalog source created is the same as the package name. (https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/53ba0b717599ec9c780df1c2ed31af9fef145467/internal/policy/operator/deployable_by_olm.go#L255)

Would it be possible to update the https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/53ba0b717599ec9c780df1c2ed31af9fef145467/internal/openshift/openshift.go#L240 to do some form of pre-validation of the catalog source name and add a prefix to it if it's not valid? 

I might be missing something but the pre-flight repository purpose seems to be for verification purposes, if the bundle is accepted it will be part of catalog source "community-operators" therefore I think we could maybe consider changing the catalog source name to some static value instead of adding a prefix/having it the same as package name.

**Pull request merged**